### PR TITLE
[Feature] Support cloud native MV with storage volume

### PR DESCRIPTION
**Pull request by TszKitLo40**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 39 / 48 (81.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/MaterializedViewHandler.java|1|2|50.00%|[769]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|31|39|79.49%|[2052, 2853, 2854, 2855, 2856, 2991, 3021, 3022]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|7|7|100.00%|[]|

**Review by HangyuanLiu — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/alter/MaterializedViewHandler.java [hunk 1]
```diff
@@ -765,6 +765,9 @@ public void processDropMaterializedView(DropMaterializedViewStmt dropMaterialize
             checkDropMaterializedView(mvName, olapTable);
             // Step2; drop data in memory
             long mvIndexId = dropMaterializedView(mvName, olapTable);
+            if (olapTable.isCloudNativeTable()) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/alter/MaterializedViewHandler.java:768]: this is old rollup, and don't care

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java [hunk 2]
```diff
@@ -1027,6 +1028,16 @@ public String getMaterializedViewDdlStmt(boolean simple) {
...
         }
 
+        if (isCloudNativeMaterializedView()) {
+            StorageVolumeMgr svm = GlobalStateMgr.getCurrentState().getStorageVolumeMgr();
+            String storageVolumeId = svm.getStorageVolumeIdOfTable(id);
+            if (storageVolumeId != null) {
+                String volume = GlobalStateMgr.getCurrentState().getStorageVolumeMgr().getStorageVolumeName(storageVolumeId);
+                sb.append(StatsConstants.TABLE_PROPERTY_SEPARATOR).append(
+                        PropertyAnalyzer.PROPERTIES_STORAGE_VOLUME).append("\" = \"").append(volume).append("\"");
+            }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java:1038]: implement this in `LakeMaterializedView.appendUniqueProperties`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 39 / 48 (81.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/MaterializedViewHandler.java|1|2|50.00%|[769]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|31|39|79.49%|[2052, 2853, 2854, 2855, 2856, 2991, 3021, 3022]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|7|7|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 3]
```diff
@@ -2823,27 +2823,39 @@ public void createMaterializedView(CreateMaterializedViewStatement stmt)
...
+                storageVolumeId = GlobalStateMgr.getCurrentState().getStorageVolumeMgr()
+                        .getStorageVolumeIdOfTable(materializedView.getId());
+            }
+            onCreate(db, materializedView, storageVolumeId, stmt.isIfNotExists());
+        } catch (DdlException e) {
+            if (materializedView.isCloudNativeMaterializedView()) {
+                GlobalStateMgr.getCurrentState().getStorageVolumeMgr().unbindTableToStorageVolume(materializedView.getId());
+                throw e;
+            }
+        }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:2858]:
```suggestion
        } catch (DdlException e) {
            if (materializedView.isCloudNativeMaterializedView()) {
                GlobalStateMgr.getCurrentState().getStorageVolumeMgr().unbindTableToStorageVolume(materializedView.getId());
            }
            throw e;
        }
```

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 4]
```diff
@@ -3102,6 +3125,9 @@ public void dropMaterializedView(DropMaterializedViewStmt stmt) throws DdlExcept
             if (refreshTask != null) {
                 taskManager.dropTasks(Lists.newArrayList(refreshTask.getId()), false);
             }
+            if (view.isCloudNativeMaterializedView()) {
+                GlobalStateMgr.getCurrentState().getStorageVolumeMgr().unbindTableToStorageVolume(view.getId());
+            }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:3130]: `Database.dropTable` already has unbind operation.
better update the check in `Database.dropTable`.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 32 / 41 (78.05%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|2|3|66.67%|[162]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|29|37|78.38%|[2052, 2853, 2854, 2855, 2856, 2991, 3021, 3022]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 44 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|29|37|78.38%|[2052, 2853, 2854, 2855, 2857, 2991, 3021, 3022]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|6|6|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 44 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|29|37|78.38%|[2052, 2853, 2854, 2855, 2857, 2991, 3021, 3022]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|6|6|100.00%|[]|

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 36 / 44 (81.82%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|29|37|78.38%|[2052, 2853, 2854, 2855, 2857, 2991, 3021, 3022]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/LakeMaterializedView.java|6|6|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Comment by TszKitLo40** [comment]
https://github.com/Mergifyio backport branch-3.1

**Pull request merged**